sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/TableChange.java

```diff
@@ -726,7 +746,16 @@ public String[] fieldNames() {
      * is not provided as updating column default values does not need to back-fill existing data.
      * Empty string means dropping the column default value.
      */
-    public String newDefaultValue() { return newDefaultValue; }
+    public String newDefaultValue() {
+      return newDefaultValue == null ? "" : newDefaultValue.getSql();
+    }
+
+    /**
+     * Returns the column default value as {@link DefaultValue}. The default value literal
+     * is not provided as updating column default values does not need to back-fill existing data.
+     * Empty string and Null literal means dropping the column default value.
+     */
+    public DefaultValue newModelDefaultValue() { return newDefaultValue; }
```

Comment:
Why not interpret null for `DefaultValue` as drop?

Reply:
Looks like for ALTER COLUMN, null string meant no default was set, it was different than "" meaning drop default.

https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala#L5246

I was trying to be consistent with that, I could take a look, but it would be more code to check if we are "" and set null.

Reply:
>  null string meant no default was set

Shouldn't we avoid generating `UpdateColumnDefaultValue` at all in that case?

Reply:
yes thats the current mechanism here:  

https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/v2AlterTableCommands.scala#L245

if its null, we dont make UpdateColumnDefaultValue.  Else we make it with ""

Reply:
@cloud-fan, do you by any chance remember the context on why we used "" instead of null?

Reply:
Discussed a bit offline, we can probably change catalyst side and add a boolean/enum to AlterColumnSpec to indicate whether its Drop or Unchanged.

In DSV2 side, we will have no UpdateColumnDefaultValue for unchanged, have UpdateColumnDefaultValue new API return null for Drop case, and have it return the default if there's actually one.

We will preserve the old UpdateColumnDefaultValue string API to return "" though for backward compatibility.

@cloud-fan if you have thoughts